lib/srv/db/redis/protocol/resp2.go

```diff
@@ -0,0 +1,205 @@
+/*
+
+ Copyright 2022 Gravitational, Inc.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+
+
+*/
+
+package protocol
+
+import (
+	"strconv"
+
+	"github.com/go-redis/redis/v8"
+	"github.com/gravitational/trace"
+)
+
+// WriteCmd writes Redis commands passed as vals to Redis wire form.
+// Most types is covered by go-redis implemented WriteArg() function. Types override by this function are:
+// * Redis errors and Go error: go-redis returns a "human-readable" string instead of RESP compatible error message
+// * integers: go-redis converts them to string, which is not always what we want to.
```

Comment:
```suggestion
// Most types are covered by go-redis implemented WriteArg() function. Types override by this function are:
// * Redis errors and Go error: go-redis returns a "human-readable" string instead of RESP compatible error message
// * integers: go-redis converts them to string, which is not always what we want.
```